### PR TITLE
feat: allow ignoring protoc version comments in go generate check

### PR DESCRIPTION
**Pull request by galargh**
Resolves https://github.com/pl-strflt/uci/issues/26

Tested in https://github.com/libp2p/go-libp2p/pull/2603. I'll create a PR to go-libp2p that enables the newly added option after I release the new patch version of uCI which will be shortly after this is merged.

### CODE REVIEW

File: .github/workflows/go-check.yml [hunk 1]
```diff
@@ -79,12 +82,23 @@ jobs:
...
           run: |
             git clean -fd # make sure there aren't untracked files / directories
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
+              find . -name '*.pb.go' -print0 | xargs -0 sed -i '/^\/\/.*protoc.*v/d'
+              git add .
+              git commit -m "chore: remove protoc version comments"
+            fi
             go generate -x ./...
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
+              find . -name '*.pb.go' -print0 | xargs -0 sed -i '/^\/\/.*protoc.*v/d'
```
Comment (galargh) [inline .github/workflows/go-check.yml:97]:
```suggestion
              find . -name '*.pb.go' -print0 | xargs -0 -r sed -i '/^\/\/.*protoc.*v/d'
```

File: .github/workflows/go-check.yml [hunk 2]
```diff
@@ -79,12 +82,23 @@ jobs:
       - name: go generate
         uses: protocol/multiple-go-modules@v1.2
         if: (success() || failure()) && fromJSON(steps.config.outputs.json).gogenerate == true
+        env:
+          IGNORE_PROTOC_VERSION_COMMENTS: ${{ inputs.go-generate-ignore-protoc-version-comments }}
         with:
           run: |
             git clean -fd # make sure there aren't untracked files / directories
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
+              find . -name '*.pb.go' -print0 | xargs -0 sed -i '/^\/\/.*protoc.*v/d'
```
Comment (galargh) [inline .github/workflows/go-check.yml:91]:
```suggestion
              find . -name '*.pb.go' -print0 | xargs -0 -r sed -i '/^\/\/.*protoc.*v/d'
```

File: .github/workflows/go-check.yml [hunk 3]
```diff
@@ -79,12 +82,23 @@ jobs:
...
         if: (success() || failure()) && fromJSON(steps.config.outputs.json).gogenerate == true
+        env:
+          IGNORE_PROTOC_VERSION_COMMENTS: ${{ inputs.go-generate-ignore-protoc-version-comments }}
         with:
           run: |
             git clean -fd # make sure there aren't untracked files / directories
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
+              find . -name '*.pb.go' -print0 | xargs -0 -r sed -i '/^\/\/.*protoc.*v/d'
+              git add .
+              git commit -m "chore: remove protoc version comments"
```
Comment (galargh) [inline .github/workflows/go-check.yml:93]:
```suggestion
              git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
              git config --global user.name "github-actions[bot]"
              git commit -m "chore: remove protoc version comments"
```

File: .github/workflows/go-check.yml [hunk 4]
```diff
@@ -79,12 +82,25 @@ jobs:
...
+          IGNORE_PROTOC_VERSION_COMMENTS: ${{ inputs.go-generate-ignore-protoc-version-comments }}
         with:
           run: |
             git clean -fd # make sure there aren't untracked files / directories
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
+              find . -name '*.pb.go' -print0 | xargs -0 -r sed -i '/^\/\/.*protoc.*v/d'
+              git add .
+              git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+              git config --global user.name "github-actions[bot]"
+              git commit -m "chore: remove protoc version comments"
```
Comment (galargh) [inline .github/workflows/go-check.yml:95]:
```suggestion
              git diff --cached --quiet || git commit -m "chore: remove protoc version comments"
```

File: .github/workflows/go-check.yml [hunk 5]
```diff
@@ -79,12 +82,25 @@ jobs:
...
             git clean -fd # make sure there aren't untracked files / directories
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
+              find . -name '*.pb.go' -print0 | xargs -0 -r sed -i '/^\/\/.*protoc.*v/d'
+              git add .
+              git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+              git config --global user.name "github-actions[bot]"
+              git diff --cached --quiet || git commit -m "chore: remove protoc version comments"
+            fi
             go generate -x ./...
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
```
Comment (galargh) [inline .github/workflows/go-check.yml:98]:
```suggestion
            if [[ "$IGNORE_PROTOC_VERSION_COMMENTS" == "true" ]]; then
```

File: .github/workflows/go-check.yml [hunk 6]
```diff
@@ -79,12 +82,25 @@ jobs:
       - name: go generate
         uses: protocol/multiple-go-modules@v1.2
         if: (success() || failure()) && fromJSON(steps.config.outputs.json).gogenerate == true
+        env:
+          IGNORE_PROTOC_VERSION_COMMENTS: ${{ inputs.go-generate-ignore-protoc-version-comments }}
         with:
           run: |
             git clean -fd # make sure there aren't untracked files / directories
+            if [[ -n "$IGNORE_PROTOC_VERSION_COMMENTS" ]]; then
```
Comment (galargh) [inline .github/workflows/go-check.yml:90]:
```suggestion
            if [[ "$IGNORE_PROTOC_VERSION_COMMENTS" == "true" ]]; then
```

File: .github/workflows/go-check.yml [hunk 7]
```diff
@@ -79,12 +82,25 @@ jobs:
...
+        env:
+          IGNORE_PROTOC_VERSION_COMMENTS: ${{ inputs.go-generate-ignore-protoc-version-comments }}
         with:
           run: |
             git clean -fd # make sure there aren't untracked files / directories
+            if [[ "$IGNORE_PROTOC_VERSION_COMMENTS" == "true" ]]; then
+              find . -name '*.pb.go' -print0 | xargs -0 -r sed -i '/^\/\/.*protoc.*v/d'
+              git add .
+              git config --global user.email "41898282+github-actions[bot]@users.noreply.github.com"
+              git config --global user.name "github-actions[bot]"
```
Comment (marten-seemann) [inline .github/workflows/go-check.yml:94]: Do I understand correctly that this won't be actually commit the repo? Is this commit only used on CI?

**Review by marten-seemann — APPROVED** [review]

File: .github/workflows/go-check.yml (hunk 7 above)
Comment (galargh) [reply .github/workflows/go-check.yml:94]: Yes, that's 100% right. We're only making a local commit to make it easier to check the diff. But even for local commits, git wants us to set user details.

**Review by laurentsenta — APPROVED** [review]

**Comment by galargh** [comment]
Merging, I'll cut a new release tomorrow

**Pull request merged**